docs/migrating-from-mdl.md

```diff
@@ -0,0 +1,318 @@
+option for complex applications that create and destroy parts of their user interface in runtime.
+
+MDC-Web modules include both ES2015 sources and bundled, transpiled ES5, so you can choose the best option depending on
+your build pipeline.
+
+#### Importing a component from ES2015 sources
+
+Start by importing the component:
+
+```js
+import {MDCTextfield} from 'mdc-textfield';
+```
+
+And instantiate a component by calling the constructor on the root node:
+
+```js
+const textfield = new MDCTextfield(document.querySelector('.mdc-textfield'));
+```
+
+Be sure to store the returned instance somewhere so that you can access the instance when you need to; unlike
+auto-initialization, there is no way to retrieve it later via the DOM.
+
+#### Using the component class in an ES5 bundle
+
+Each MDC-Web component ships with an ES5 transpiled [UMD](https://github.com/umdjs/umd) bundle, with component classes placed into a package-specific property inside of the `mdc` namespace. In order to easily access it in your code, you can do:
+
+```js
+const MDCTextfield = mdc.textfield.MDCTextfield;
+```
+
+After that, you can instantiate a component by calling the constructor on the root node:
+
+```js
+const textfield = new MDCTextfield(document.querySelector('.mdc-textfield'));
+```
+
+Be sure to store the returned instance somewhere so that you can access the instance when you need to; unlike
+auto-initialization, there is no way to retrieve it later via the DOM.
+
+
+## Styling
+
+Styling in MDL is achieved with a collection of CSS classes that get applied to the DOM. Internally, MDL is built with
+Sass, but there was no effort in exposing the Sass mixins and functions to developers.
+
+In MDC-Web, components can expose both CSS and Sass interfaces to their styles. This is particularly true for
+foundational components, such as `@material/elevation` or `@material/theme`.
+
+### Using CSS classes
+
+Like in MDL, styling components with CSS classes is simply a matter of applied them to the DOM:
+
+```html
+<div class="mdc-textfield">
+  <input type="text" id="my-textfield" class="mdc-textfield__input">
+  <label class="mdc-textfield__label" for="my-textfield">Hint text</label>
+</div>
+```
+
+Every component lists the required CSS classes, as well as all of the optional modifiers, as part of their README.
+
+### Using CSS Custom Properties
+
+Some MDC-Web components include CSS custom properties as part of their interfaces. An example of this is the
+`@material/layout-grid`, which exposes custom properties for modifying the default margin and gutter sizes:
+
+```css
+.my-grid {
+  --mdc-layout-grid-margin: 40px;
+  --mdc-layout-grid-gutter: 16px;
+}
+```
+
+Be sure to read the README for each component to get all the details on their usage.
+
+> **Note:** CSS Custom Property support in components is a progressive enhancement feature that doesn’t work in every supported browser. If
```

Comment:
Should we talk about how we also usually provide SCSS variables / plain CSS styling guidance in order to help with this?